core/API/DataTablePostProcessor.php

```diff
@@ -150,15 +150,15 @@ public function applyPivotByFilter(DataTableInterface $dataTable)
         $pivotBy = Common::getRequestVar('pivotBy', false, 'string', $this->request);
         if (!empty($pivotBy)) {
             $this->applyComputeProcessedMetrics($dataTable);
+            $dataTable = $this->convertSegmentValueToSegment($dataTable);
 
-            $reportId = $this->apiModule . '.' . $this->apiMethod;
             $pivotByColumn = Common::getRequestVar('pivotByColumn', false, 'string', $this->request);
             $pivotByColumnLimit = Common::getRequestVar('pivotByColumnLimit', false, 'int', $this->request);
 
+            $dataTable->filter('PivotByDimension', array($this->report, $pivotBy, $pivotByColumn, $pivotByColumnLimit,
+                PivotByDimension::isSegmentFetchingEnabledInConfig()));
             $dataTable->filter('ColumnCallbackDeleteMetadata', array('segmentValue'));
             $dataTable->filter('ColumnCallbackDeleteMetadata', array('segment'));
```

Comment:
Is there a point in calling `convertSegmentValueToSegment() ` when we remove the segment afterwards?

Reply:
Yes, the segment is used by PivotByDimension (useful for supporting UserCountry.City for example, whose segment involves country & region),.